File: Engine/DocGear.cs
using HtmlAgilityPack;
using ChileTrabajos.Models;
using System.Globalization;

namespace ChileTrabajos.Engine
{
    public class DocGear
    {
        public List<JobItem> ListJobs(string srtHtml)
        {
            List<JobItem> jobsWithLinks = AddLinks(srtHtml);
            List<JobItem> jobsWithDates = AddDate(jobsWithLinks,srtHtml);

            for (int i = 0; i < jobsWithLinks.Count; i++)
            {
                jobsWithLinks[i].CalendarDate = jobsWithDates[i].CalendarDate;
            }

            return jobsWithLinks;
        }

        public List<JobItem> AddLinks(string srtHtml)
        {
            HtmlDocument doc = new HtmlDocument();;
            doc.LoadHtml(srtHtml);
            HtmlNodeCollection jobItems = doc.DocumentNode.SelectNodes("//div[contains(@class, 'job-item')]");
            List<JobItem> jobList = new List<JobItem>();

            foreach (var jobItem in jobItems)
            {
                string link = jobItem.SelectSingleNode(".//a").GetAttributeValue("href", string.Empty);
                jobList.Add(new JobItem { UrlLink = link });
            }

            return jobList;
        }

        public List<JobItem> AddDate(List<JobItem> jobList, string srtHtml)
        {
            HtmlDocument doc = new HtmlDocument();
            doc.LoadHtml(srtHtml);
            HtmlNodeCollection jobItems = doc.DocumentNode.SelectNodes("//div[contains(@class, 'job-item')]//h3[2]");
            List<string> dates = new List<string>();

            foreach (var jobItem in jobItems)
            {
                string text = jobItem.SelectSingleNode(".//a").InnerText;
                dates.Add(text.Trim().Replace("  "," "));
            }
            
            for(int i = 0; i < jobList.Count ;i++){
                jobList[i].CalendarDate = DateTime.ParseExact(dates[i],"d 'de' MMMM 'de' yyyy",new CultureInfo("es-ES"));;    
            }
            return jobList;
        }
    }
}

File: Modules/Engine/DocGear.py
import locale
from datetime import datetime
from bs4 import BeautifulSoup

from ..Models.JobItem import JobItem

# Configurar el locale a español
locale.setlocale(locale.LC_TIME, 'es_ES.UTF-8')  # Esto puede variar según tu sistema operativo

def list_jobs(srt_html):
    """
    return List[JobItem]
    """    
    jobs_with_links = add_links(srt_html)
    jobs_with_dates = add_date(jobs_with_links, srt_html)
    for i in range(len(jobs_with_links)):
        jobs_with_links[i].calendar_date = jobs_with_dates[i].calendar_date
    return jobs_with_links

def add_links(srt_html):
    soup = BeautifulSoup(srt_html, 'html.parser')
    job_items = soup.select('div.job-item')
    job_list = []
    for job_item in job_items:
        link = job_item.select_one('a')['href']
        job_list.append(JobItem(link))
    return job_list

def add_date(job_list, srt_html):
    soup = BeautifulSoup(srt_html, 'html.parser')
    job_items = soup.select('div.job-item h3:nth-of-type(2)')
    dates = []
    for job_item in job_items:
        text = job_item.select_one('a').text
        dates.append(text.strip().replace('  ', ' '))
    for i in range(len(job_list)):
        job_list[i].calendar_date = datetime.strptime(dates[i], "%d de %B de %Y")
    return job_list
